src/storage/txn/process.rs

```diff
@@ -506,8 +505,8 @@ fn notify_deadlock_detector_if_needed(
     is_pessimistic_txn: bool,
     txn_ts: u64,
 ) {
-    if detector_scheduler.is_some() && is_pessimistic_txn {
-        detector_scheduler.as_ref().unwrap().clean_up(txn_ts);
+    if let (Some(detector_scheduler), false) = (detector_scheduler.as_ref(), is_pessimistic_txn) {
```

Comment:
The `is_pessimistic_txn` should be `true`.

Reply:
Good catch!